mono/mini/genmdesc.c

```diff
@@ -220,7 +220,7 @@ init_table (void) {
 
 static void
 output_char (FILE *f, char c) {
-	if (isalnum (c))
+	if (isalnum ((unsigned char)c)) // Cast to unsigned char to work around a bug in MSVCRT in debug builds
```

Comment:
We *are* getting a value < 1 and I would like to know why. The `OpDesc` structure in genmdesc.c has a `spec` field. The `build_table` function then calls `output_char` on exactly six characters from this field. `output_char` prints the character if it's alphanumeric, or a `\x` hex code if it is not. I see no indication in the program what `spec` is or what its allowable values are and I am not sure where it comes from. When I ran the program, `spec` contained binary gibberish-- the first byte was 0, at least one other byte was 0, and at least one of the bytes (a latin-1 euro symbol as it happens...) happened to have the high order bit set. Because output_char takes c as a signed char this byte wound up being negative and therefore triggers the assert in Microsoft's `isalnum`.

So: Regardless of whether the Microsoft behavior is a bug or not, casting to unsigned char is the correct behavior for `output_char`. The function expects the range 0-255 and so we force the value into that range.

However, it seems probable to me that the `spec` field containing `'\0'`s and high-order ASCII in the first place indicates a bug somewhere else. It is hard to tell without knowing what `spec` does.